Tests/Negotiation/FormatNegotiatorTest.php

```diff
@@ -103,6 +103,45 @@ public function testGetBestWithFormat()
         $this->assertEquals(new Accept('application/json;version=1.0'), $this->negotiator->getBest(''));
     }
 
+    public function testGetBestWithPreferExtension()
+    {
+        $priorities = ['text/html', 'application/json'];
+        $this->addRequestMatcher(true, ['priorities' => $priorities, 'prefer_extension' => '2.0']);
+
+        $reflectionClass = new \ReflectionClass(get_class($this->request));
```

Comment:
I wanted to do that, but Request->getRequestUri() caches the result, so any tests afterwards would erroneously use the first request uri set. Actually it should be using "pathInfo" because getPathInfo also caches its result and that is what is used by FormatNegotiator. I'll push an update to fix that.